lib/modules/dosomething/dosomething_reportback/includes/Reportback.php

```diff
@@ -121,12 +121,22 @@ public static function find(array $filters = []) {
       throw new Exception('No reportback data found.');
     }
 
-    foreach($results as $item) {
-      // @TODO: remove need for passing variable for constructor check.
-      $reportback = new static(['ignore' => TRUE]);
-      $reportback->build($item);
+    if (isset($filters['load_user_data'])) {
```

Comment:
I don't think you need the `load_user_data` check here, which forces you to repeat the code in both the `if` and then the `else`. I would recommend putting it in the `build()` function.